streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -617,6 +617,10 @@ public void shouldRespectCommitNeeded() {
 
         task.commit();
         assertFalse(task.commitNeeded());
+
+        // confirm that timestamp was correctly committed
+        assertTrue(Long.parseLong(task.consumer.committed(partition1).metadata())
+                   == task.getPartitionTime(partition1));
```

Comment:
Can we add separate unit tests to confirm this produces the expected behavior? I think the JIRA had some examples highlighting why this is a problem, it would be good to convert those into tests to make sure we're really fixing the problem at hand :)

Reply:
No problem. Added a new test case to confirm behavior.